### DISCUSSION TITLE
The Future of Client-Side Transitions and Core Web Vitals

### DISCUSSION THREAD

**Discussion by cohandy** [vercel/next.js#21607] [Help]
My company has recently started to adopt Next.js and we were excited to link our pages together via soft navigations to vastly improve our UX and page speeds. We then heard of the announcement from Google about [adding page experience to Google search](https://webmasters.googleblog.com/2020/11/timing-for-page-experience.html) so we began work on optimizing our pages Core Web Vitals.

We initially thought that activating soft transitions would give us a major advantage in LCP timings, until we [dug in more](https://twitter.com/rick_viscomi/status/1316756074524016648) and found out that Google doesn't count soft transitions in Chrome UX reporting (what Google will use to determine page experience). Meaning that client side transitions using `next/Link` _will not factor in at all_ to Core Web Vitals. 

How is everyone dealing with this? Our website is extremely reliant on SEO and we are at the whim of Google with this upcoming update. We feel that we shouldn't even activate soft transitions as those page views won't count. Although we are considering using something like the [network information api](https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API) to determine users on 3g or lower and activating soft transitions for them (so we'd altogether eliminate slow traffic). 

We've also found it difficult to even find information on this topic. It'd be great if you all could share any links or information you've found.

**Comment by gravelcycles** [comment]
Here's another thread in `GoogleChrome/web-vitals` from a member of Google on why SPAs currently do not have Web Vitals support on soft page transitions.

https://github.com/GoogleChrome/web-vitals/issues/119#issuecomment-767298992
